src/contexts/bridge.context.tsx

```diff
@@ -568,14 +588,110 @@ const BridgeProvider: FC<PropsWithChildren> = (props) => {
+      const isPendingDepositInApiBridges = (depositTxHash: string) => {
+        return bridges.find((bridge) => {
+          return (
+            (bridge.status === "initiated" || bridge.status === "on-hold") &&
+            bridge.depositTxHash === depositTxHash
+          );
+        });
+      };
+      const isPendingClaimInApiBridges = (claimTxHash: string) => {
+        return bridges.find((bridge) => {
+          return bridge.status === "completed" && bridge.claimTxHash === claimTxHash;
+        });
+      };
+
+      return Promise.all(
+        pendingTxs.map(async (pendingTx) => {
+          if (
+            pendingTx.status === "pending-deposit" &&
+            isPendingDepositInApiBridges(pendingTx.depositTxHash)
+          ) {
+            return storage.removePendingTx(pendingTx.depositTxHash);
+          }
+          if (
+            pendingTx.status === "pending-claim" &&
+            isPendingClaimInApiBridges(pendingTx.claimTxHash)
+          ) {
+            return storage.removePendingTx(pendingTx.depositTxHash);
+          }
+
+          const txHash =
+            pendingTx.status === "pending-deposit"
+              ? pendingTx.depositTxHash
+              : pendingTx.claimTxHash;
+          const provider =
+            pendingTx.status === "pending-deposit"
+              ? pendingTx.from.provider
+              : pendingTx.to.provider;
+          const tx = await provider.getTransaction(txHash);
+
+          if (isTxCanceled(tx)) {
+            return storage.removePendingTx(pendingTx.depositTxHash);
+          }
+
+          if (isTxMined(tx)) {
+            const txReceipt = await provider.getTransactionReceipt(txHash);
+
+            if (hasTxBeenReverted(txReceipt)) {
+              return storage.removePendingTx(pendingTx.depositTxHash);
+            }
+          }
+
+          if (Date.now() > pendingTx.timestamp + PENDING_TX_TIMEOUT) {
+            return storage.removePendingTx(pendingTx.depositTxHash);
+          }
+        })
+      ).then();
+    },
+    [env]
+  );
+
+  const getPendingBridges = useCallback(
+    async (bridges?: Bridge[]): Promise<PendingBridge[]> => {
+      if (bridges) {
+        await cleanPendingTxs(bridges);
+      }
+
+      if (!env) {
+        throw new Error("Env is not available");
+      }
+
+      return Promise.all(
+        storage.getPendingTxs(env).map(async (tx) => {
+          const chain = env?.chains.find((chain) => chain.networkId === tx.networkId);
+          const fiatAmount =
+            chain && env.fiatExchangeRates.areEnabled
+              ? await getTokenPrice({ token: tx.token, chain })
+              : undefined;
+
+          return {
+            status: "pending",
+            networkId: tx.networkId,
+            from: tx.from,
+            to: tx.to,
+            depositTxHash: tx.depositTxHash,
+            claimTxHash: tx.status === "pending-claim" ? tx.claimTxHash : undefined,
+            token: tx.token,
```

Comment:
We should probably make use of `addWrappedToken` here to make sure it's complete.

Reply:
Right, done!